### ISSUE TITLE
support for globs in `watch_file`

### ISSUE THREAD

**Issue by gekoke** [direnv/direnv#1263]
Does `watch_file` support file globs?

If yes, then this should be documented under the [watch_file section in the manual](https://direnv.net/man/direnv-stdlib.1.html#codewatchfile-ltpathgt-ltpathgt-code).

If no, then I think this would be a sensible (almost essential?) feature to have.

**Comment by zimbatm** [comment]
yes, globs are built into bash, so they work everywhere. You can think of globs like a pre-processing phase that expands the list of arguments passed to the program or function.

**Comment by gekoke** [comment]
Ah, I didn't realize .envrc files were Just Bash Scripts™, since I really only use them to load flake devShells - that explains it. I think a few examples using globs in the docs wouldn't hurt, though!

**Comment by commiterate** [comment]
Semi-related to this: would it make sense for `watch_dir` to be variadic (i.e. `watch_dir <dir> [<dir> ...]`)?

Right now we can use glob patterns for `watch_file` since it accepts multiple file paths.

We can't, however, for `watch_dir` since it only accepts 1 directory path.

For example, assume we have a file system layout like this:

```
.
├── inputs
│   │   # Main (default) overlay.
│   ├── main
│   │   └── nix
│   │       └── packages
│   │           └── «system»
│   │               ├── mypackage.nix
│   │               ├── mypackage_1.nix
│   │               └── mypackage_2.nix
│   │
│   │   # Test overlay.
│   └── test
│       └── nix
│           ├── checks
│           │   └── «system»
│           │       └── default.nix
│           │
│           └── devShells
│               └── «system»
│                   └── default.nix
│
├── .envrc
│   # Using packagesFromDirectoryRecursive to create attribute sets from file system layouts.
├── flake.nix
└── flake.lock
```

This `.envrc` will only watch `inputs/main/nix` since it's the first item returned by glob expansion.

`.envrc`

```bash
watch_dir inputs/*/nix

use flake
```

We can work around this with a Bash for loop but it would be convenient if `watch_dir` were variadic.
